experimental/structs.md

```diff
@@ -0,0 +1,55 @@
+# [JavaScript Structs: Fixed Layout Objects and Some Synchronization Primitives](proposal-structs)
+
+# Structs
+
+```js
+interface Struct <: Class {
```

Comment:
Is it correct to extend from class? I'm not sure it is considering shared structs are limited in what they allow (eg no private members)

Reply:
Quoting the readme for shared structs:

> They cannot contain instance methods or instance private names.

Non-shared structs just say they share their syntax with classes. I haven't had a chance to read the spec itself though.

Reply:
I believe Brad is right. As I understand it, Struct is more of a _sibling_ to Class than a _descendent_ - because of Brad's quoted note on non-shared syntax.

As a novice to TC39 proposals, I found the AST naming a bit confusing in that the general "Struct" is less restricted than the more specific "SharedStruct". I found it easier to think of them as named "UnsharedStruct" and "SharedStruct" per the spec text.

Putting it together, the way I'm understanding the hierarchy is roughly:

* `Class` and `SharedStruct` extend `Declaration`
* (`Unshared`)`Struct` extends `Class`, `SharedStruct`

Is that roughly accurate?

Reply:
> They cannot contain instance methods or instance private names.

Yes, a share construct can not contain non-constructor instance methods/instance private names, however, they could contain static methods/static private names ([spec](https://tc39.es/proposal-structs/#sec-struct-definitions-static-semantics-containsinstancemethod)), which are modelled as MethodDefinition/PropertyDefinition with `static: true`. Although the shared struct is more limited than the non-shared struct, they do share the same element types. So I added the `shared: boolean` to model both of them.

Note that whether the interface `Struct` extends from `Class` is mostly editorial because they are not the final AST shape: They are intermediate interface that helps us structure the ESTree spec. If we agree that `Struct` should be considered a sibling of `Class`, I can revert the last commit.

Reply:
>  If we agree that Struct should be considered a sibling of Class, I can revert the last commit.

Yeah if we do that I'd just use a base interface mostly to ensure that we don't accidentally diverge in property names between the structs and classes.

Reply:
Okay, I should confirm the editorial convention here: If we extract a new base interface, let's say `ClassOrStruct` (better names are welcomed), should I directly change the `es2015.md` in this PR? Or should I put those changes in the `structs.md`?

Reply:
Just in `structs.md`, yeah. So you'd have something like `extend interface Class <: ClassOrStruct {}` followed by declaration for struct itself.